flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/kryo/KryoSerializerClassLoadingTest.java

```diff
@@ -63,8 +63,8 @@ void restoreOriginalClassLoader() {
     void guardTestAssumptions() {
         assertThatThrownBy(
                         () -> Class.forName(OUTSIDE_CLASS_LOADING.getObject().getClass().getName()))
-                .isInstanceOf(ClassNotFoundException.class)
-                .withFailMessage("This test's assumptions are broken");
+                .withFailMessage("This test's assumptions are broken")
```

Comment:
We'd better put the fail message to the parameter of `assertThatThrownBy` as this message actually means that this code block should throw an exception, not just about whether the type match.

For example: 
```
assertThatThrownBy(
                        () -> Class.forName(OUTSIDE_CLASS_LOADING.getObject().getClass().getName()), "This test's assumptions are broken")
```